### ISSUE TITLE
strict export declare modifiers rules applies incorrectly to declarations in tests

### ISSUE THREAD

**Issue by sandersn** [Microsoft/dtslint#209]
```ts
export declare class C {
  constructor(x: number);
}
```

Expected: Legal in a test (.ts) file.
Actual: Error saying that I should just use `export`, which is wrong.

I'm not convinced this rule should apply to tests at all; ideally DT contributors should be able to paste in whatever example code they can find from documentation or tutorials. Making it conform to strict linting discourages that.
